### PR TITLE
Issue 1680

### PR DESCRIPTION
**Pull request by oddodaoddo**
**Description of the Pull Request (PR):**

Write your description of the PR here. Be sure to include as much background,
and details necessary for the reviewers to understand exactly what this is
fixing or enhancing.

**This fixes or addresses the following GitHub issues:**

- Ref: #

**Checkoff for all PRs:**

- [x] I have read the [Guidelines for Contributing](https://github.com/singularityware/singularity/blob/master/ and this PR conforms to the stated requirements.
- [x ] I have added changes to the [CHANGELOG](https://github.com/singularityware/singularity/blob/development/CHANGELOG.md) and and documentation updates to the [singularityware](https://www.github.com/singularityware/singularityware.github.io) documentation base.
- [x] I have tested this PR locally with a `make test`
- [x ] This PR is NOT against the project's master branch
- [x ] I have added myself as a contributor to the [contributors's file](https://github.com/singularityware/singularity/blob/master/CONTRIBUTORS.md)
- [x ] This PR is ready for review and/or merge

Attn: @singularityware-admin

### CODE REVIEW

File: src/docker-extract.c [hunk 1]
```diff
@@ -417,13 +425,39 @@ int main(int argc, char **argv) {
...
+    
+    if (statvfs(layer_target_fs, &stat) != 0)
+	 fs_avail_space = -1;
+    else
+	fs_avail_space = stat.f_bsize * stat.f_bavail;
+
+    if (fs_avail_space == -1)
+	singularity_message(WARNING, "Unable to obtain free space on %s\n", layer_target_fs);
+
+    singularity_message(DEBUG, "Available /tmp space %lu bytes\n", fs_avail_space);
```
Comment (jmstover) [inline src/docker-extract.c]: Where `/tmp` is hard coded (and below), you should use `layer_target_fs` for the target... like:

`singularity_message(DEBUG, "Available %s space %lu bytes\n", layer_target_fs, fs_avail_space);`

**Comment by oddodaoddo** [comment]
Sorry about that - I get excited about my first PR to be potentially accepted - I should pay more attention. Anyway, fixed and I re-ran make test and tested also in a VM with a restricted size /tmp...

Ognen

File: src/docker-extract.c [hunk 2]
```diff
@@ -243,7 +249,7 @@ int extract_tar(const char *tarfile, const char *rootfs_dir) {
     char *orig_dir;
     const char *pathname;
     int pathtype;
-
+
```
Comment (dtrudg) [inline src/docker-extract.c]: whitespace

File: src/docker-extract.c [hunk 3]
```diff
@@ -417,13 +425,39 @@ int main(int argc, char **argv) {
...
+    retval = apply_whiteouts_and_get_tar_size(tarfile, rootfs_realpath, &exploded_tar_size);
 
     if (retval != 0) {
         singularity_message(ERROR, "Error applying layer whiteouts\n");
         ABORT(255);
     }
 
+    // following logic is to make sure we have enough space on filesystem to explode layers of image
+    if (!exploded_tar_size)
+	singularity_message(WARNING, "Unable to estimate expanded size of archive %s or size is zero.\n", tarfile);
```
Comment (dtrudg) [inline src/docker-extract.c]: there's a mix of tabs and spaces for indentation in this area - please could you reconcile? Should be a 4 space indent in this file.

**Review by dtrudg — CHANGES_REQUESTED** [review]
Code looks good to me. Just some really picky comments r.e. getting the indentation consistent.

File: src/docker-extract.c (hunk 3 above)
Comment (oddodaoddo) [reply src/docker-extract.c]: Thanks! My primary machine is FreeBSD and singularity doesn't run on it ;-) so I develop in a Linux VM and I think my vi tab settings are all messed up. I sent a fix, hopefully that does it.

**Review by dtrudg — APPROVED** [review]

**Pull request closed without merging**